### PR TITLE
Feat/processor inputs outputs

### LINKED ISSUE #493: Improved workflow step display
We want to improve the workflow step display.

The target goal is to develop a whole graph, but as a first step we can, keep that flat organization and improve it to prepare it:

1. Be able to distinguish raw_data_processor from sample_group_processors: different color, larger boxes with a header to easily make the difference.
2. Add input and output colored markers for what is produced and consumed.

### PR DESCRIPTION
**Pull request by bertrandboudaud**
Added inputs and output items to the workflow steps.

Also added Current Session Node on top (at the moment, it shows all possible outputs, even if the data there is empty: could be a next step to display it only when data is there)

Improved a little bit the dragging node handling.

InputOutputs

### CODE REVIEW

File: src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp [hunk 1]
```diff
@@ -37,6 +37,16 @@
 
 namespace SmartPeak
 {
+  std::set<std::string> CalculateCalibration::getInputs() const
+  {
+    return { "Features"};
```
Comment (dmccloskey) [inline src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp]: "Standards Concentrations" and "Quantitation Methods" should also be inputs I believe.

File: src/smartpeak/source/core/ApplicationHandler.cpp [hunk 2]
```diff
@@ -174,6 +174,18 @@ namespace SmartPeak
...
+  std::set<std::string> ApplicationHandler::Command::getInputs() const
+  {
+    const auto description = getIProcessorDescription();
+    return (description ? description->getInputs() : std::set<std::string>());
+  }
+
+  std::set<std::string> ApplicationHandler::Command::getOutputs() const
+  {
+    const auto description = getIProcessorDescription();
+    return (description ? description->getOutputs() : std::set<std::string>());
```
Comment (dmccloskey) [inline src/smartpeak/source/core/ApplicationHandler.cpp:186]: Nice one liner.

File: src/smartpeak/source/core/RawDataProcessors/FilterFeatures.cpp [hunk 3]
```diff
@@ -35,6 +35,16 @@
 namespace SmartPeak
 {
 
+  std::set<std::string> FilterFeatures::getInputs() const
+  {
+    return { "Feature Filters", "Targeted Experiment" };
```
Comment (dmccloskey) [inline src/smartpeak/source/core/RawDataProcessors/FilterFeatures.cpp]: "Features" is missing I think

**Review by dmccloskey — CHANGES_REQUESTED** [review]
The updates to the workflow widget look very nice.  I found a couple of missing inputs in the `getInputs` methods that I noted inline.

File: src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp (hunk 1 above)
Comment (bertrandboudaud) [reply src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp]: indeed. thanks

**Review by dmccloskey — APPROVED** [review]

**Pull request merged**